graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```diff
@@ -281,19 +287,33 @@ abstract class BaseDataTypeGenerator(
 
         val fieldSpecs = javaType.build().fieldSpecs
         fieldSpecs.forEachIndexed { index, fieldSpec ->
-            if (fieldSpec.type.isPrimitive) {
-                equalsBody.append("${fieldSpec.name} == that.${fieldSpec.name}")
-            } else {
-                equalsBody.append("java.util.Objects.equals(${fieldSpec.name}, that.${fieldSpec.name})")
+            // Only include data fields for comparison, exclude is<field>Set Boolean fields
+            var field = fields.find { iter -> ReservedKeywordSanitizer.sanitize(iter.name) == fieldSpec.name }
+            if (field != null) {
+                if (fieldSpec.type.isPrimitive) {
+                    equalsBody.append(
+                        """
+                            
+                            ${fieldSpec.name} == that.${fieldSpec.name} 
+                        """.trimIndent()
+                    )
+                } else {
+                    equalsBody.append(
+                        """
+                            
+                            java.util.Objects.equals(${fieldSpec.name}, that.${fieldSpec.name}) 
+                        """.trimIndent()
+                    )
+                }
+                if (index != fieldSpecs.size - 1) {
+                    equalsBody.append("&&")
+                }
             }
+        }
 
-            if (index != fieldSpecs.size - 1) {
-                equalsBody.append(
-                    """ &&
-                    
-                    """.trimMargin()
-                )
-            }
+        if (equalsBody.endsWith('&')) {
```

Comment:
Instead of doing this, you can just use trimSuffix("&&") and convert to string directly in line 323: 
```
methodBuilder.addStatement(equalsBody.removeSuffix("&&").toString())
```

Reply:
Thanks, updated